sdk/src/main/java/io/dapr/client/DaprHttp.java

```diff
@@ -132,7 +126,7 @@ public int getStatusCode() {
    * @return Asynchronous text
    */
   public Mono<Response> invokeAPI(String method, String urlString, Map<String, String> urlParameters, byte[] content, Map<String, String> headers) {
-    return Mono.fromFuture(CompletableFuture.supplyAsync(
```

Comment:
I've checked with Azure SDK team, let's keep the CompletableFuture (old code) but wrap it with Mono.defer(). This way, we still have a separate threadpool to do the http call and not block on the mono caller's thread.

Reply:
using fromCallable will do the same

Reply:
But you will not have a separate threadpool, you would be using the caller's thread instead.

Reply:
What's the point of having a separate thread, if we still need to wait for the thread to be completed?

Reply:
To show the benefits we would need to do performance testing. This change is better already because it is fixing a functionality bug, so I will take it. We might end up bringing back the threadpool but that will be separate.